pypeit/extraction.py

```diff
@@ -410,11 +410,12 @@ def extract(self, global_sky, bkg_redux_global_sky=None, model_noise=None, spat_
         if len(remove_idx) > 0:
             self.sobjs.remove_sobj(remove_idx)
 
-        # Add the DETECTOR container, S/N ratio, and FWHM in ARCSEC for each extracted object
+        # Add the DETECTOR container, S/N ratio, FWHM and BOX_RADIUS in ARCSEC for each extracted object
         for sobj in self.sobjs:
             sobj.DETECTOR = self.sciImg.detector
             sobj.S2N = sobj.med_s2n()
-            sobj.SPAT_FWHM = sobj.med_fwhm()
+            sobj.SPAT_FWHM = sobj.med_fwhm(sobj.platescale(spectrograph=self.spectrograph))
+            sobj.SPAT_BOX_RADIUS = sobj.boxcar_arcsec(sobj.platescale(spectrograph=self.spectrograph))
```

Comment:
I feel like `sobj` should know about the spectrograph.  Can we add it?

Reply:
Hi @kbwestfall . I added `PYP_SPEC` here and a sobj property that gets the spectrograph. I wish I could add `PYP_SPEC` earlier in the code (like during obj finding) but it looks too complicated to do it without increasing the chances to fail.
What do you think?